docs/architecture/adrs/007-orchestrator-telemetry.md

```diff
@@ -0,0 +1,74 @@
+
+### Instrumentation
+An incoming request to the orchestrator will initialize a new trace, therefore a trace-id and request should be in
+one-to-one correspondence. All important functions throughout the control flow of handling a request in the orchestrator 
+will be instrumented with the `#[tracing::instrument]` attribute macro. This will create and enter a span for each 
+function call and add it to the trace of the request. Here, important functions refers to any function that performs
+important business logic or that may incur significant latency, this includes all the handler functions for incoming
+and outgoing requests.
+
+### Metrics
+The orchestrator will aggregate metrics regarding the requests it has received/handled, and annotate the metrics with
+span attributes allowing for detailed filtering and monitoring. The metrics will be exported through the OTLP exporter
+through the metrics provider. Traces exported through the traces provider can also have R.E.D. (request, error and
+duration) metrics attached to them implicitly by the OTEL collector using the `spanmetrics` connector. Both the OTLP
+metrics and the `spanmetrics` metrics can be exported to configured metrics backends like Prometheus or Grafana.
+The orchestrator will handle a variety of useful metrics such as counters and histograms for received/handled 
+successful/failed requests, request and stream durations, and server/client errors.
+
+### Configuration
+The orchestrator will expose CLI args/env variables for configuring the OTLP exporter endpoint(s) 
+(e.g. `OTEL_EXPORTER_OTLP_ENDPOINT=http://localhost:4317`) and protocol(s) (e.g. `OTEL_EXPORTER_OTLP_PROTOCOL=grpc`), 
+as well as the ability to enable trace (includes logs) and metrics OTLP exporting (e.g. `OTLP_EXPORT=traces,metrics`.
+Alongside this, the orchestrator will also allow for configuring the Rust log level (e.g. `RUST_LOG=debug`) and 
+log format to use for logs printed to stdout (e.g. `STDOUT_LOG=pretty`). Printing logs to stdout can also be disabled
+(e.g. `STDOUT_LOG=none`). The OTEL collector can be configured to receive from the OTLP endpoint(s) and export to
+observability backends. Examples of a complete configuration for the orchestrator and OTEL collector can be found in
+the playbooks [LINK HERE] and [LINK HERE] as Docker Compose files and as an OpenShift deployment.
+
+### Cross-service tracing
+The orchestrator and client services will be able to consolidate telemetry and share observability through a common
+configuration and backends. This will be made possible through the use of the OTLP standard as well as through the
+propagation of the trace context through requests across services using the standardized `traceparent` header. The
+orchestrator will make use of an incoming request's `traceparent` for initializing a trace context for a request if
+it exists, or else will initialize a new trace. 
```

Comment:
Its not super clear from above, but we will be passing this `traceparent` to downstream via headers as well right ?

Reply:
Can you clarify what you mean by downstream? 

Also this was a question I've been meaning to bring up, I already know traceparent will be sent back and forth from orchestrator to configured clients, but Im less certain whether the traceparent propagates into/out of the orchestrator in user requests and responses, if you could confirm. It sounds like something that would be beneficial (so a user could set up telemetry around all their services and orchestrator's, in other words the orchestrator would get the trace id for a new incoming request from its traceparent header if present and create a new trace if its missing, and it will pass a traceparent back in the response) but I wanted to make sure this isnt somehow leaking more information than should be back to the user (the same sort of concern one has with error messages and not returning debug details that the user shouldnt be privvy to)